.github/workflows/_integration_tests.yml

```diff
@@ -97,22 +97,30 @@ jobs:
       fail-fast: false
       matrix:
         test: [
+          direct-curl-portal-down,
           direct-curl-portal-restart,
+          direct-curl-portal-relay-down,
+          relayed-curl-portal-down,
           relayed-curl-portal-restart,
           relayed-curl-relay-restart,
-          direct-curl-portal-down,
-          relayed-curl-portal-down,
-          direct-curl-portal-relay-down,
-          direct-download-roaming-network,
-          dns-etc-resolvconf,
           dns-nm,
           dns-failsafe, # Uses the default DNS control method
           systemd/dns-systemd-resolved,
         ]
         include:
           - test: direct-download-roaming-network
             dns-control: etc-resolv-conf
-          - test: dns-etc-resolvconf
+          - test: direct-dns-etc-resolvconf
+            dns-control: etc-resolv-conf
+          - test: direct-dns-etc-resolvconf-portal-down
+            dns-control: etc-resolv-conf
+          - test: direct-dns-etc-resolvconf-relay-down
+            dns-control: etc-resolv-conf
+          - test: relayed-dns-etc-resolvconf
+            dns-control: etc-resolv-conf
+          - test: relayed-dns-etc-resolvconf-portal-down
+            dns-control: etc-resolv-conf
+          - test: relayed-dns-etc-resolvconf-relay-restart
             dns-control: etc-resolv-conf
```

Comment:
I guess `dns-control` is here because the scripts don't bring docker-compose up, so we need to tell the Github Action to use `/etc/resolv.conf` DNS control, even though the script redundantly mentions `/etc/resolv.conf` in its filename.

I don't think the separated setup steps do anything for us in general, they're just a workaround for writing code inside YAML. If we replace the shell scripts with Python (or something better), then I'd like to merge the setup step in too. Then the DNS control can just be another parameter and it won't have to be passed twice.

Reply:
Yeah, I'll clean this up

Reply:
Simplified